pkg/skaffold/build/cache/hash_test.go

```diff
@@ -88,6 +88,36 @@ func TestGetHashForArtifact(t *testing.T) {
 				},
 			},
 			expected: "09b366c764d0e39f942283cc081d5522b9dde52e725376661808054e3ed0177f",
+		}, {
```

Comment:
We should add tests to make sure the hash is not equal for a artifacts with different build args and  Two identical artifact should yield the same hash.

Reply:
The current tests should cover both of those:

`build args`
`build args in different order`
`different build args`

Reply:
umm no, we don't explicitly test if the artifact has is different if everything else is same. We are only checking for value. 
In future, if someone changes code here (and by mistake disregards build args), they might just change `expected` to match what was actual. It does not guard against breaking this in future.

Can we move all these cases to `TestArtifactConfig` where you explicitly compute hash on 2 artifacts and check if artifact hash is not equal when artifacts is same type but 
1. Build args different
2. Build args are same. 
3. Build args are same but different order

Reply:
Yup, ended up writing a new test, `TestBuildArgs`. Thanks!